src/renderer/components/TeamList.jsx

```diff
@@ -84,32 +72,39 @@ export default class TeamList extends React.PureComponent {
         this.setState({indexToRemoveServer: -1});
     }
 
-    render() {
-        const self = this;
-        const teamNodes = this.props.teams.map((team, i) => {
-            function handleTeamRemove() {
-                document.activeElement.blur();
-                this.openServerRemoveModal(i);
-            }
-
-            function handleTeamEditing() {
-                document.activeElement.blur();
-                this.handleTeamEditing(team.name, team.url, i, team.order);
-            }
+    handleTeamRemovePrompt = (index) => {
+        return () => {
+            document.activeElement.blur();
+            this.openServerRemoveModal(index);
+        };
+    }
 
-            function handleTeamClick() {
-                this.props.onTeamClick(team.name);
-            }
+    handleTeamEditing = (team, index) => {
+        return () => {
+            document.activeElement.blur();
+            this.setState({
+                showEditTeamForm: true,
+                team: {
+                    url: team.url,
+                    name: team.name,
+                    index,
+                    order: team.order,
+                },
+            });
+        };
+    }
 
+    render() {
+        const teamNodes = this.props.teams.map((team, i) => {
             return (
                 <TeamListItem
                     index={i}
                     key={`teamListItem_${team.name}`}
```

Comment:
seems these two are not used by the component, we can remove them

Reply:
Have to keep `key` to satisfy linter but we can remove `index`.